Svc/DpCatalog/DpCatalog.cpp

```diff
@@ -308,9 +318,13 @@ namespace Svc {
 
     void DpCatalog::appendFileState(const DpStateEntry& entry) {
         FW_ASSERT(this->m_stateFileData);
+        FW_ASSERT(entry.dir < static_cast<FwIndexType>(this->m_numDirectories),
+            entry.dir,
+            static_cast<FwAssertArgType>(this->m_numDirectories)
+        );
 
         // We will append state to the existing state file
-        // FIXME: Have to handle case where state file has partially transmitted
+        // TODO: Have to handle case where state file has partially transmitted
```

Comment:
Remaining TODO.

Reply:
That's on purpose, since this is still technically a prototype.